cardano-node/Cardano/Node/CLI.hs

```diff
@@ -93,34 +118,67 @@ data SomeProtocol where
   SomeProtocol :: (RunDemo blk, TraceConstraints blk)
                => Consensus.Protocol blk -> SomeProtocol
 
-fromProtocol :: Protocol -> IO SomeProtocol
-fromProtocol BFT =
+fromProtocol :: CardanoConfiguration -> Protocol -> IO SomeProtocol
+fromProtocol _ BFT =
     case Consensus.runProtocol p of
       Dict -> return $ SomeProtocol p
   where
     p = ProtocolMockBFT defaultSecurityParam
-fromProtocol Praos =
+fromProtocol _ Praos =
     case Consensus.runProtocol p of
       Dict -> return $ SomeProtocol p
   where
     p = ProtocolMockPraos defaultDemoPraosParams
-fromProtocol MockPBFT =
+fromProtocol _ MockPBFT =
     case Consensus.runProtocol p of
       Dict -> return $ SomeProtocol p
   where
     p = ProtocolMockPBFT defaultDemoPBftParams
-fromProtocol RealPBFT =
+fromProtocol CardanoConfiguration{ccCore=ccCore@Core{coStaticKeyMaterial}} RealPBFT = do
+    let Genesis{geSrc, geGenesisHash} = coGenesis ccCore
+        genHash = either (throw . ConfigurationError) id $ -- TODO: need a proper error tag
+                  decodeAbstractHash geGenesisHash
+        cvtRNM :: RequireNetworkMagic -> RequiresNetworkMagic
+        cvtRNM NoRequireNetworkMagic = RequiresNoMagic
+        cvtRNM RequireNetworkMagic   = RequiresMagic
+        StaticKeyMaterial{skmSigningKeyFile, skmDlgCertFile} = coStaticKeyMaterial
+        kmoDeserialiseDelegateKey = Signing.SigningKey . snd . either (error . show) id . deserialiseFromBytes Signing.fromCBORXPrv
+    geneConfig <- either (error . show) id <$> -- TODO: need a proper error tag
+      runExceptT (Genesis.mkConfigFromFile (cvtRNM $ coRequiresNetworkMagic ccCore) geSrc genHash)
+
+    sk  <- kmoDeserialiseDelegateKey <$> LB.readFile skmSigningKeyFile
+    dlg <- either (error . show) id . CanonicalJSON.canonicalDecPre <$> LB.readFile skmDlgCertFile
+
+    let plc = PbftLeaderCredentials sk dlg
+        p   = ProtocolRealPBFT defaultDemoPBftParams geneConfig (pure plc)
+
     case Consensus.runProtocol p of
-      Dict -> return $ SomeProtocol p
-  where
-    p = ProtocolRealPBFT defaultDemoPBftParams genesisConfig
-    genesisConfig = Dummy.dummyConfig
+      Dict -> do
+        return . SomeProtocol $ p
```

Comment:
This can probably revert to `Dict -> return $ SomeProtocol p`

Reply:
Done!